spring-cloud-netflix-hystrix/src/main/java/org/springframework/cloud/netflix/hystrix/security/HystrixSecurityAutoConfiguration.java

```diff
@@ -56,18 +57,33 @@ public void init() {
 				.getPropertiesStrategy();
 		HystrixCommandExecutionHook commandExecutionHook = HystrixPlugins.getInstance()
 				.getCommandExecutionHook();
+		HystrixConcurrencyStrategy concurrencyStrategy = getConcurrencyStrategy();
 
 		HystrixPlugins.reset();
 
 		// Registers existing plugins excepts the Concurrent Strategy plugin.
 		HystrixPlugins.getInstance().registerConcurrencyStrategy(
-				new SecurityContextConcurrencyStrategy(existingConcurrencyStrategy));
+				new SecurityContextConcurrencyStrategy(concurrencyStrategy));
 		HystrixPlugins.getInstance().registerEventNotifier(eventNotifier);
 		HystrixPlugins.getInstance().registerMetricsPublisher(metricsPublisher);
 		HystrixPlugins.getInstance().registerPropertiesStrategy(propertiesStrategy);
 		HystrixPlugins.getInstance().registerCommandExecutionHook(commandExecutionHook);
 	}
 
+	private HystrixConcurrencyStrategy getConcurrencyStrategy() {
+		HystrixConcurrencyStrategy registeredStrategy = HystrixPlugins.getInstance()
+				.getConcurrencyStrategy();
+		//Hystrix Registered a default Strategy.
+		//if registeredStrategy not the default and not some with existingConcurrencyStrategy throw a exception.
+		if (registeredStrategy != null
+				&& !registeredStrategy.equals(HystrixConcurrencyStrategyDefault.getInstance())
```

Comment:
Could we just check if its an `instanceof`?

Reply:
Thanks for you good advice, it more clearer. I add test file and change  `HystrixSecurityAutoConfiguration#getConcurrencyStrategy` for easy to read and  understand. Hope to hear advice from you.